### ISSUE TITLE
Setup a public conversation channel in Gitter

### ISSUE THREAD

**Issue by oleg-nenashev** [jenkinsci/configuration-as-code-plugin#100]
It would help other contributors to participate in the project

- [x] Create a channel
- [x] Setup GitHub integration there
- [ ] Add Gitter badge to the repository

**Comment by oleg-nenashev** [comment]
@daniel-beck Why 👎 ? Currently there is a private channel in use, Gitter definitely makes it better?

**Comment by batmat** [comment]
I don't feel *that* strongly about the Gitter channel, though IMO IRC should be fine. Either on #jenkins, or by creating a dedicated new #jenkins-conf-as-code or something if judged better.

But I do feel more strongly about the fact there should already not be any private channel for this project. Things should be discussed in public as much as possible (I understand the need for videoconferencing to move forward, not saying there shouldn't be *any* private things). But even those video meetings should ideally be held in a public fashion, as much as possible, so that anyone can chime in.

Also... We have generally been strongly discouraging using GitHub issues for consistency with other plugins and core. Linking things together is easier when everything is in the Jenkins JIRA, etc.

**Comment by oleg-nenashev** [comment]
> I don't feel that strongly about the Gitter channel, though IMO IRC should be fine. Either on #jenkins, or by creating a dedicated new #jenkins-conf-as-code or something if judged better.

IRC offers a pretty bad UX for conversations when you have no custom clients like IRCCloud. Things like echelog are not fine in 2018 IMHO. There are other Jenkins subprojects using Gitter (e.g. BlueOcean), so I do not see much harm in it

> Also... We have generally been strongly discouraging using GitHub issues for consistency with other plugins and core. Linking things together is easier when everything is in the Jenkins JIRA, etc.

I totally agree with that. I said the same to the CasC team at FOSDEM. OTOH having GitHub tickets for boilerplate changes before the first alpha release is fine IMO. Once it goes to alpha, JIRA should become the main issue tracker.

**Comment by ewelinawilkosz** [comment]
yes, that's what we've agreed on - up to this stage github issues were enough for us.
but we know we may need to introduce JIRA at some point, soon...

Gitter is only an easier way to communicate around CasC about something that we don't have/need issue for

**Comment by buep** [comment]
> not be any private channel for this project

For the reference it was a CloudBees and Praqma private channel to coordinate the effort early in kicking this project of. I saw there was discussed to many issues there, so did @oleg-nenashev so we moved to a public channel.

I'll keep the private channel for a while, mostly because we can there discuss how many hours Praqma can fund etc. and things related to our customers that might be early movers.

But I very much favor we keep things in public, and even more we keep most relevant information in issues if related to such.
Or alternative someone takes action summarizing relevant discussion on an issues after some conclusion if it ended up on a chat thread instead.


(Not commenting on the Github vs Jira discussion .... I'll follow the crowd, but also notices a lot of the plugins currently out there are moving to Github issues.... we did the same with all our plugins).
